lib/mongoHandler.js

```diff
@@ -144,10 +144,26 @@ MongoStore.prototype.initialise = function(resourceConfig) {
   }
   self.resourceConfig = resourceConfig;
   self.relationshipAttributeNames = MongoStore._getRelationshipAttributeNames(resourceConfig.attributes);
-  mongodb.MongoClient.connect(self._config.url).then(function(db) {
+  mongodb.MongoClient.connect(self._config.url, {
+    server: {
+      reconnectTries: 999999999,
+      reconnectInterval: 5000
+    }
+  }).then(function(db) {
     self._db = db;
+    self._db.on("close", function(err) {
+      console.error("mongodb connection closed:", err.message);
+      self.ready = false;
+      self._db.collection("Nope").findOne({ _id: 0 }, { _id: 0 }, function() {
```

Comment:
How about simply [listing](http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#listCollections) the db's collections to do the polling? Might be a bit more efficient. Not sure. Either way is fine.